certbot/compat/filesystem.py

```diff
@@ -44,6 +44,39 @@ def chmod(file_path, mode):
         _apply_win_mode(file_path, mode)
 
 
+# One could ask why there is no copy_ownership() function, or even a reimplementation
+# of os.chown() that would modify the ownership of file without touching the mode itself.
+# This is because on Windows, it would require recalculating the existing DACL against
+# the new owner, since the DACL is composed of ACEs that targets a specific user, not dynamically
+# the current owner of a file. This action would be necessary to keep consistency between
+# the POSIX mode applied to the file and the current owner of this file.
+# Since copying and editing arbitrary DACL is very difficult, and since we actually know
+# the mode to apply at the time the owner of a file should change, it is easier to just
+# change the owner, then reapply the known mode, as copy_ownership_and_apply_mode() does.
+def copy_ownership_and_apply_mode(src, dst, mode, copy_user, copy_group):
+    # type: (str, str, int, bool, bool) -> None
+    """
+    Copy ownership (user and optionally group) from the source to the destination,
+    then apply given mode in compatible way for Linux and Windows. This replaces
+    the os.chown command.
```

Comment:
I was just manually testing this function and noticed that `copy_group` has no effect on Windows. This is totally understandable, but I think we should document this here and/or in the documentation for the parameter down below.

Reply:
Yes, I added it to the doc.